src/luarocks/repos.lua

```diff
@@ -208,7 +208,7 @@ local function move_suffixed(from_file, to_file, suffix)
 
    suffix = suffixed_from_file:sub(#from_file + 1)
    local suffixed_to_file = to_file .. suffix
-   return fs.move(suffixed_from_file, suffixed_to_file)
+   return fs.move(suffixed_from_file, suffixed_to_file, "read")
```

Comment:
these calls to fs.move should not have "read" too (they are also used copy executables). Just removing these arguments will suffice once the changes to fs.copy discussed above are in place